stan/math/prim/mat/fun/cholesky_decompose.hpp

```diff
@@ -5,6 +5,10 @@
 #include <stan/math/prim/mat/err/check_pos_definite.hpp>
 #include <stan/math/prim/mat/err/check_square.hpp>
 #include <stan/math/prim/mat/err/check_symmetric.hpp>
+#ifdef STAN_OPENCL
+#include <stan/math/gpu/cholesky_decompose.hpp>
+#include <algorithm>
```

Comment:
Why do you need to include this here? is this for `copy`? Should probably include `algorithm` in our `copy` header and use that instead, right?

Reply:
> is this for copy?
Yeah I'm pretty sure this is because cpplint complains when it sees copy without the algorithm header (even if the call is `stan::math::copy()`). Can we change that in cpplint?

Reply:
Yes, the ccplint complained with the error message to add <algorithm>.

Reply:
boo.

Reply:
@seantalts is this cool? I can look later to see if there is a way to get that check for copy to only happen if it's not from `stan::math`